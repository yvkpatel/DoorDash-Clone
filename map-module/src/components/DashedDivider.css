.dashed-divider {
    border-bottom: 0.3px dotted black;
    opacity: 0.5;
}
