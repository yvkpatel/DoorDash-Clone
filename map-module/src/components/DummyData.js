export const getRestaurantName = () => {
    const name = 'Resto Beirut';
    return name;
};

export const getOrderNumber = () => {
    const orderNum = '16899';
    return orderNum;
};

export const getOrderItems = () => {
    const orderItems = [
        {
            quantity: '2',
            name: 'Chicken Shawerma',
            price: '$9.87',
        },
        {
            quantity: '3',
            name: 'Chicken Shawerma Poutine',
            price: '$16.87',
        },
    ];
    return orderItems;
};

export const getSubTotal = () => {
    const subTotal = '$26.74';
    return subTotal;
};

export const getDeliveryFee = () => {
    const deliveryFee = '$3.99';
    return deliveryFee;
};

export const getTax = () => {
    const tax = '$5.64';
    return tax;
};

export const getTip = () => {
    const tip = '$5.00';
    return tip;
};

export const getTotal = () => {
    const total = '$41.37';
    return total;
};

export const getCustomerLocation = () => {
    const customerLocation = "Memorial University St. John's NL";
    return customerLocation;
};
