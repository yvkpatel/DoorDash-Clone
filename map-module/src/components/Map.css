.main {
    width: 75%;
    height: 100vh;
}
