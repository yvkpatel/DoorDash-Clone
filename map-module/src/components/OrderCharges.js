import React from 'react';
import './OrderItems.css';

import { SolidDivider, DashedDivider } from '.';

import {
    getSubTotal,
    getDeliveryFee,
    getTax,
    getTip,
    getTotal,
} from './DummyData.js';

export function OrderCharges() {
    return (
        <div>
            <p className="order-items">
                Subtotal <span>{getSubTotal()}</span>
            </p>
            <SolidDivider />
            <p className="order-items">
                Delivery Fee <span>{getDeliveryFee()}</span>
            </p>
            <p className="order-items">
                Tax <span>{getTax()}</span>
            </p>
            <p className="order-items">
                Tip <span>{getTip()}</span>
            </p>
            <DashedDivider />
            <p className="order-items">
                Total <span>{getTotal()}</span>
            </p>
            <DashedDivider />
        </div>
    );
}
