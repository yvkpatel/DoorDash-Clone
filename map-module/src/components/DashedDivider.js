import React from 'react';
import './DashedDivider.css';

export function DashedDivider() {
    return <div className="dashed-divider"></div>;
}
