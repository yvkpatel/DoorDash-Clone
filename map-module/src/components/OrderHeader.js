import React from 'react';
import { getRestaurantName, getOrderNumber } from './DummyData.js';

export function OrderHeader() {
    return (
        <div>
            <p>{getRestaurantName()}</p>
            <p>Order #{getOrderNumber()}</p>
        </div>
    );
}
