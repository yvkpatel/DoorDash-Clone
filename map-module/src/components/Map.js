import React from 'react';
import './Map.css';

import {
    GoogleMap,
    LoadScript,
    DirectionsRenderer,
    DirectionsService,
} from '@react-google-maps/api';

import { getRestaurantName, getCustomerLocation } from './DummyData.js';

// Hardcoded options to display a route on our map
const options = {
    origin: getRestaurantName(),
    destination: getCustomerLocation(),
    travelMode: 'DRIVING',
};

// Since were using React, I added a react google maps API wrapper.
export class Map extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            directionResult: null,
        };
    }

    // This is where the map is being inserted into our module.
    // It's just the starting code from here:
    // "https://react-google-maps-api-docs.netlify.app/#section-getting-started"
    render() {
        return (
            <div className="MapDiv">
                <LoadScript googleMapsApiKey={process.env.REACT_APP_API_KEY}>
                    <GoogleMap
                        options={{
                            styles: mapStyle,
                        }}
                        mapContainerStyle={containerStyle}
                        center={myLatLng}
                        zoom={15}
                    >
                        <DirectionsService
                            options={options}
                            callback={(result) => {
                                if (!this.state.directionResult) {
                                    this.setState({
                                        directionResult: result,
                                    });
                                }
                            }}
                        />
                        <DirectionsRenderer
                            directions={this.state.directionResult}
                        />
                    </GoogleMap>
                </LoadScript>
            </div>
        );
    }
}

// Map Size
const containerStyle = {
    width: '100%',
    height: '100vh',
    position: 'relative',
};

// Map Marker
const myLatLng = {
    lat: 47.56356087536938,
    lng: -52.71567239382679,
};

//Map Styling (Retro from https://mapstyle.withgoogle.com/)
const mapStyle = [
    {
        elementType: 'geometry',
        stylers: [
            {
                color: '#ebe3cd',
            },
        ],
    },
    {
        elementType: 'labels.text.fill',
        stylers: [
            {
                color: '#523735',
            },
        ],
    },
    {
        elementType: 'labels.text.stroke',
        stylers: [
            {
                color: '#f5f1e6',
            },
        ],
    },
    {
        featureType: 'administrative',
        elementType: 'geometry.stroke',
        stylers: [
            {
                color: '#c9b2a6',
            },
        ],
    },
    {
        featureType: 'administrative.land_parcel',
        elementType: 'geometry.stroke',
        stylers: [
            {
                color: '#dcd2be',
            },
        ],
    },
    {
        featureType: 'administrative.land_parcel',
        elementType: 'labels.text.fill',
        stylers: [
            {
                color: '#ae9e90',
            },
        ],
    },
    {
        featureType: 'landscape.natural',
        elementType: 'geometry',
        stylers: [
            {
                color: '#dfd2ae',
            },
        ],
    },
    {
        featureType: 'poi',
        elementType: 'geometry',
        stylers: [
            {
                color: '#dfd2ae',
            },
        ],
    },
    {
        featureType: 'poi',
        elementType: 'labels.text.fill',
        stylers: [
            {
                color: '#93817c',
            },
        ],
    },
    {
        featureType: 'poi.park',
        elementType: 'geometry.fill',
        stylers: [
            {
                color: '#a5b076',
            },
        ],
    },
    {
        featureType: 'poi.park',
        elementType: 'labels.text.fill',
        stylers: [
            {
                color: '#447530',
            },
        ],
    },
    {
        featureType: 'road',
        elementType: 'geometry',
        stylers: [
            {
                color: '#f5f1e6',
            },
        ],
    },
    {
        featureType: 'road.arterial',
        elementType: 'geometry',
        stylers: [
            {
                color: '#fdfcf8',
            },
        ],
    },
    {
        featureType: 'road.highway',
        elementType: 'geometry',
        stylers: [
            {
                color: '#f8c967',
            },
        ],
    },
    {
        featureType: 'road.highway',
        elementType: 'geometry.stroke',
        stylers: [
            {
                color: '#e9bc62',
            },
        ],
    },
    {
        featureType: 'road.highway.controlled_access',
        elementType: 'geometry',
        stylers: [
            {
                color: '#e98d58',
            },
        ],
    },
    {
        featureType: 'road.highway.controlled_access',
        elementType: 'geometry.stroke',
        stylers: [
            {
                color: '#db8555',
            },
        ],
    },
    {
        featureType: 'road.local',
        elementType: 'labels.text.fill',
        stylers: [
            {
                color: '#806b63',
            },
        ],
    },
    {
        featureType: 'transit.line',
        elementType: 'geometry',
        stylers: [
            {
                color: '#dfd2ae',
            },
        ],
    },
    {
        featureType: 'transit.line',
        elementType: 'labels.text.fill',
        stylers: [
            {
                color: '#8f7d77',
            },
        ],
    },
    {
        featureType: 'transit.line',
        elementType: 'labels.text.stroke',
        stylers: [
            {
                color: '#ebe3cd',
            },
        ],
    },
    {
        featureType: 'transit.station',
        elementType: 'geometry',
        stylers: [
            {
                color: '#dfd2ae',
            },
        ],
    },
    {
        featureType: 'water',
        elementType: 'geometry.fill',
        stylers: [
            {
                color: '#b9d3c2',
            },
        ],
    },
    {
        featureType: 'water',
        elementType: 'labels.text.fill',
        stylers: [
            {
                color: '#92998d',
            },
        ],
    },
];
