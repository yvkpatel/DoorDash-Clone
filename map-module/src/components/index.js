import { DashedDivider } from './DashedDivider';
import { SolidDivider } from './SolidDivider';
import { Sidepanel } from './Sidepanel';
import { Map } from './Map';
import { OrderCharges } from './OrderCharges';
import { OrderHeader } from './OrderHeader';
import { OrderInfoTitle } from './OrderInfoTitle';
import { OrderItems } from './OrderItems';

export {
    DashedDivider,
    Map,
    OrderCharges,
    OrderHeader,
    OrderInfoTitle,
    OrderItems,
    SolidDivider,
    Sidepanel
};
