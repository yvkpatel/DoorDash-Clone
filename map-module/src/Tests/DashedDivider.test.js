import React from 'react';
import renderer from 'react-test-renderer';
import { DashedDivider } from '../components';

it('renders Dashed Divider component correctly', () => {
    let tree = renderer.create(<DashedDivider />).toJSON();
    expect(tree).toMatchSnapshot();
});
