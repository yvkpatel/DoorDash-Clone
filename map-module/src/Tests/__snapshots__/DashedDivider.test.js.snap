// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashed Divider component correctly 1`] = `
<div
  className="dashed-divider"
/>
`;
